crates/mod-crypto/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+        fields.add_field_method_get("base32hex_nopad", |_, this| {
+            Ok(BASE32HEX_NOPAD.encode(&this.0))
+        });
+
+        fields.add_field_method_get("base64", |_, this| Ok(BASE64.encode(&this.0)));
+        fields.add_field_method_get("base64_nopad", |_, this| Ok(BASE64_NOPAD.encode(&this.0)));
+
+        fields.add_field_method_get("base64url", |_, this| Ok(BASE64URL.encode(&this.0)));
+        fields.add_field_method_get("base64url_nopad", |_, this| {
+            Ok(BASE64URL_NOPAD.encode(&this.0))
+        });
+
+        fields.add_field_method_get("bytes", |lua, this| lua.create_string(&this.0));
+    }
+
+    fn add_methods<M: UserDataMethods<Self>>(methods: &mut M) {
+        methods.add_meta_method(MetaMethod::ToString, |_, this, _: ()| {
+            Ok(HEXLOWER.encode(&this.0))
+        });
+    }
+}
+
+fn aes_decrypt_block(ciphertext_buf: &[u8], params: AesParams) -> anyhow::Result<DecryptResult>  {
+    let mut in_out_buffer = ciphertext_buf.to_vec();
+    let dec_key: PaddedBlockDecryptingKey;
+
+    match params.algorithm {
+        AesAlgo::Ecb() => {
+            dec_key = match params.key {
+                AesKey::Aes128(k) => {
+                    let key = UnboundCipherKey::new(&AES_128, &k)?;
+                    PaddedBlockDecryptingKey::ecb_pkcs7(key)?
+                }
+                AesKey::Aes256(k) => {
+                    let key = UnboundCipherKey::new(&AES_256, &k)?;
+                    PaddedBlockDecryptingKey::ecb_pkcs7(key)?
+                }
+            };
+
+            match dec_key.decrypt(&mut in_out_buffer, DecryptionContext::None) {
+                //     Ok(DigestResult(digest.as_ref().to_vec()))
+                Ok(plaintext) => Ok(DecryptResult(plaintext.to_vec())),
+                Err(_) => Err(anyhow!("Decryption failed with AES key, mode ecb block")),
+            }
+        }
+
+        AesAlgo::Cbc() => {
+            dec_key = match params.key {
+                AesKey::Aes128(k) => {
+                    let key = UnboundCipherKey::new(&AES_128, &k)?;
+                    PaddedBlockDecryptingKey::cbc_pkcs7(key)?
+                }
+                AesKey::Aes256(k) => {
+                    let key = UnboundCipherKey::new(&AES_256, &k)?;
+                    PaddedBlockDecryptingKey::cbc_pkcs7(key)?
+                }
+            };
+
+            // CBC expects the IV to be prepended to the ciphertext.
+            if ciphertext_buf.len() < 16 {
+                return Err(anyhow!(
+                    "Ciphertext is too short to contain an IV (expected at least 16 bytes for IV)"
+                ));
+            }
+
+            let (iv_bytes, actual_ciphertext) = ciphertext_buf.split_at(16); // Split into IV and actual ciphertext
+            let iv: [u8; 16] = iv_bytes
+                .try_into()
+                .map_err(|_| anyhow!("IV length is incorrect, expected 16 bytes"))?;
+
```

Comment:
Lots of "magic number" usage here with `16`. Is there a constant that we can reference from the underlying crypto crate instead?

If not, let's define something for ourselves:

```rust
const CBC_IV_LEN: usize = 16;
```

then we can use it throughout:

```suggestion
            anyhow::ensure!(ciphertext_buf.len() >= CBC_IV_LEN,
                    "Ciphertext is too short to contain an IV (expected at least {CBC_IV_LEN} bytes for IV)"
                );
                
            let (iv_bytes, actual_ciphertext) = ciphertext_buf.split_at(CBC_IV_LEN); // Split into IV and actual ciphertext
            let iv: [u8; CBC_IV_LEN] = iv_bytes
                .try_into()
                .map_err(|_| anyhow!("IV length is incorrect, expected {CBC_IV_LEN} bytes"))?;

```